Issue/10651 material notification details

khaykov opened the pull request:
Part of #10651

#11192 needs to be merged first.

This PR moves notification details to the material theme and adds dark mode support.

Notifications are tricky, and I'm not sure about visual changes :) 

![comparison_notes](https://user-images.githubusercontent.com/728822/73602106-e025ce00-4522-11ea-97c6-9754535cbfec.png)

To test:
- Check details of different types of notifications and make sure they look ok, and nothing stands out as weird, in both light and dark themes.

You can change the app theme from the App Settings.


PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

jkmassel commented:
We're freezing `14.2` today, so this PR is being bumped to `14.3`. If you'd like it to ship with `14.2`, please merge it into `release/14.2` and ping me – I'll be happy to cut a new beta release.

jd-alexander commented:
Hi @khaykov good work on pushing this forward 💯 

Here a few issues I found on a Pixel 3a XL, API 29.

Mention 1 | Mention 2 
--------|-------
<img src="https://user-images.githubusercontent.com/1509205/74492050-92eb1a00-4e9b-11ea-8713-78abdbe350e1.png" width="320">     |  <img src="https://user-images.githubusercontent.com/1509205/74492053-941c4700-4e9b-11ea-98ab-d018980c2993.png" width="320">

Comment 1 | Comment 2 
--------|-------
<img src="https://user-images.githubusercontent.com/1509205/74492054-94b4dd80-4e9b-11ea-80f1-6761556585fc.png" width="320">        |   <img src="https://user-images.githubusercontent.com/1509205/74492055-954d7400-4e9b-11ea-8409-0938678dab3b.png" width="320">

Badge | Comment 3 
--------|-------
<img src="https://user-images.githubusercontent.com/1509205/74492056-954d7400-4e9b-11ea-88ef-1d8e4d57ad88.png" width="320">        |  <img src="https://user-images.githubusercontent.com/1509205/74492060-95e60a80-4e9b-11ea-85d1-c56c278374af.png" width="320">

jd-alexander commented on WordPress/src/main/java/org/wordpress/android/ui/notifications/blocks/CommentUserNoteBlock.java:
```diff
@@ -256,10 +255,11 @@ public void configureResources(Context context) {
-        mNormalBackgroundColor = context.getResources().getColor(android.R.color.white);
-        mAgoTextColor = context.getResources().getColor(R.color.neutral);
-        mUnapprovedTextColor = context.getResources().getColor(R.color.warning_60);
+
```
I think this white line can be removed.

mattmiklic commented:
It might be simpler to keep the bottom border, which would be fine too.

<img width="792" alt="Screen Shot 2020-02-14 at 3 20 40 PM" src="https://user-images.githubusercontent.com/1396552/74564676-9931d280-4f3d-11ea-9d8b-8e689bf0d03a.png">

mattmiklic commented:
Let's try Yellow 20 in both light and dark mode, because the darker versions of yellow get muddy and hard to discern from green if you're colorblind.

renanferrari commented on WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NoteBlockAdapter.java:
```diff
@@ -48,8 +46,6 @@ public View getView(int position, View convertView, ViewGroup parent) {
         convertView.setTag(R.id.note_block_tag_id, noteBlock.getBlockType());
 
-        noteBlock.setBackgroundColor(mBackgroundColor);
-
```
Since we're removing this, couldn't we remove the `mBackgroundColor` field and its setter from `NoteBlock` as well?

khaykov replied:
Good catch! We can safely remove it :)

renanferrari commented on WordPress/src/main/res/layout/note_block_user.xml:
```diff
@@ -1,84 +1,78 @@
 
         <LinearLayout
-            android:layout_width="match_parent"
+            android:layout_width="fill_parent"
```
Same thing as before: shouldn't we use `match_parent` instead of `fill_parent`?

```suggestion
            android:layout_width="match_parent"
```

khaykov replied:
`fill_parent` is a blast from the past for sure :)

renanferrari commented on WordPress/src/main/res/layout/note_block_footer.xml:
```diff
@@ -24,21 +23,21 @@
 
         <org.wordpress.android.widgets.WPTextView
             android:id="@+id/note_footer_text"
             android:layout_width="fill_parent"
```
Shouldn't we use `match_parent` instead of `fill_parent`?

```suggestion
            android:layout_width="match_parent"
```

khaykov commented:
@mattmiklic Updated the color! I also noticed that the border on your screenshots is a bit thicker, so I adjusted the size accordingly. Let me know if it's not intended :) 

![comparison_nt](https://user-images.githubusercontent.com/728822/74581669-baf37f80-4f66-11ea-80f5-1275c7450d25.png)

@renanferrari @jd-alexander  Thanks for the review! I addressed your concerns :)

jd-alexander commented on WordPress/src/main/res/drawable/bg_rectangle_warning_surface_with_padding.xml:
```diff
@@ -2,21 +2,21 @@
-            <solid android:color="@color/warning" />
+            <solid android:color="?attr/wpColorWarningLight" />
             <padding
                 android:bottom="1dp"
```
Is there a reason these values were hardcoded? I know it's outside the scope of this PR to modify this so they could be left as is. Let me know.

khaykov replied:
Those values are a "hackish" way to play with layers to get the required drawable. I don't think they should be exposed in dimens, since describing them in a variable name without confusing someone non-familiar with them is hard. I exposed `notification_comment_padded_frame_bg_offset` and `notification_comment_frame_bg_border_width` that you can play with to change the drawable border.

jd-alexander replied:
Okay understood! Makes sense 🙏
